src/posit/connect/resources.py

```diff
@@ -61,77 +61,108 @@ def __init__(self, ctx: Context, path: str, pathinfo: str = "", /, **attributes)
         ctx : Context
             The context object containing the session and URL for API interactions.
         path : str
-            The HTTP path component for the collection endpoint
-        pathinfo : str
-            The HTTP part of the path directed at a specific resource
+            The HTTP path component for the resource endpoint
         **attributes : dict
             Resource attributes passed
-
-        Attributes
-        ----------
-        _ctx : Context
-            The context object containing the session and URL for API interactions
-        _path : str
-            The HTTP path for the collection endpoint.
         """
         params = ResourceParameters(ctx.session, ctx.url)
         super().__init__(params, **attributes)
         self._ctx = ctx
-        self._path = posixpath.join(path, pathinfo)
+        self._path = path
 
 
 T = TypeVar("T", bound="Active")
 """A type variable that is bound to the `Active` class"""
 
 
 class ActiveSequence(ABC, Generic[T], Sequence[T]):
-    def __init__(self, ctx: Context, path: str, pathinfo: str = "", uid: str = "guid"):
-        """A sequence abstraction for any HTTP GET endpoint that returns a collection.
+    """A sequence for any HTTP GET endpoint that returns a collection."""
 
-        It lazily fetches data on demand, caches the results, and allows for standard sequence operations like indexing and slicing.
+    _cache: Optional[List[T]]
```

Comment:
```suggestion
    _cache: List[T] | None
```

Reply:
For my own understanding: What's the difference between these two?

Reply:
Technically nothing. They both resolve to `T | None`. 

I am on the stance of `Optional` should only be used as a parameter type as the parameter is "optional". But for typing, tell me it can have a `None` type.

Reply:
Unfortunately we have to use the explicit `Optional` and `Union` types in all cases for return types. The shorthand Union type operator was introduced in Python 3.10 and hasn't been backported to 3.9 or 3.8.

In 3.8, changing to `T | None` throws ` TypeError: unsupported operand type(s) for |: 'TypeVar' and 'NoneType'`.

This isn't the case for argument types. I don't know why yet, but it just works. My guess is that argument types aren't evaluated by the interpretation and Pyright is smart enough to deal with that fact. For example https://github.com/posit-dev/posit-sdk-py/blob/ac488c731d4a456555ef2ed6102cfaa3ca7b4d84/src/posit/connect/content.py#L320-L321 🤷🏻 

Side note; I've always thought of it as the inverse. Accept `T | None` and return `Optional[T]`. Or, in natural language, this method accepts `T` or `None` and sometimes returns `T`. Which stems from my background in Java, where the `T | None` syntax doesn't exist.

Reply:
By requiring the use of 

```
from __future__ import annotations
```

We can use `|` syntax on _not top level_ usage. 

```
Bad = int | str

class Ok:
    def method(self) -> int | str: ...
```

(This check is now made within ruff and will give a good suggested fix 😄 )